internal/handlers/common/error.go

```diff
@@ -158,7 +171,89 @@ func formatBitwiseOperatorErr(err error, operator string, maskValue any) error {
 	}
 }
 
+// WriteErrors represents slice of protocol write errors.
+type WriteErrors []WriteError
+
+// Error implements error interface.
+func (we *WriteErrors) Error() string {
+	var err string
+	for _, e := range *we {
+		err += e.Error() + ","
+	}
+
+	return err
+}
+
+// Code returns error code.
+func (we *WriteErrors) Code() ErrorCode {
+	for _, e := range *we {
+		return e.code
+	}
+	return errUnset
+}
+
+// Unwrap implements standard error unwrapping interface.
+func (we *WriteErrors) Unwrap() error {
+	for _, e := range *we {
+		return &e
+	}
+	return nil
+}
+
+// Document returns wire protocol error document.
+func (we WriteErrors) Document() *types.Document {
```

Comment:
Why other methods are implemented on the pointer, but that one is one the value?